commit 6a32ac11f9377350a4608d3318b7b7a2dee691b7 (refs/changes/35/1035/3)
Author: Yang Yong <yy1111.yang@samsung.com>
Date:   2016-12-22T11:21:23+09:00 (2 years, 10 months ago)
    
    joyent/node-triton#148 triton profile edit failes if EDITOR contains a space

diff --git a/lib/common.js b/lib/common.js
index 93f3b68..8a61f1e 100644
--- a/lib/common.js
+++ b/lib/common.js
@@ -16,6 +16,7 @@ var getpass = require('getpass');
 var os = require('os');
 var path = require('path');
 var read = require('read');
+var spawnargs = require('spawn-args');
 var strsplit = require('strsplit');
 var tty = require('tty');
 var util = require('util'),
@@ -790,7 +791,20 @@ function editInEditor(opts, cb) {
 
     // TODO: want '-f' opt for vi? What about others?
     var editor = process.env.EDITOR || '/usr/bin/vi';
-    var kid = child_process.spawn(editor, [tmpPath], {stdio: 'inherit'});
+    editor = editor.trim();
+
+    var editorArgs;
+    var optIdx = editor.indexOf(' ');
+    if (optIdx === -1) {
+        editorArgs = [tmpPath];
+    } else {
+        var editorOpt = editor.substr(optIdx, editor.length);
+        editorArgs = spawnargs(editorOpt + ' ' + tmpPath);
+
+        editor = editor.substr(0, optIdx);
+    }
+
+    var kid = child_process.spawn(editor, editorArgs, {stdio: 'inherit'});
     kid.on('exit', function (code) {
         if (code) {
             return (cb(code));
diff --git a/package.json b/package.json
index c96f85b..cdd10ef 100644
--- a/package.json
+++ b/package.json
@@ -22,6 +22,7 @@
     "rimraf": "2.4.4",
     "semver": "5.1.0",
     "smartdc-auth": "2.5.2",
+    "spawn-args": "0.2.0",
     "sshpk": "1.10.1",
     "sshpk-agent": "1.4.2",
     "strsplit": "1.0.0",
